test/OpenTelemetry.Tests/Shared/VerifyNoEventSourceErrorsLoggedTestAttribute.cs

```diff
@@ -0,0 +1,47 @@
+// <copyright file="VerifyNoEventSourceErrorsLoggedTestAttribute.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Tests.Shared
+{
+    using System.Diagnostics.Tracing;
+    using System.Linq;
+    using System.Reflection;
+    using Xunit;
+    using Xunit.Sdk;
+
+    /// <summary>
+    /// Can be used to assert that no Errors were logged to EventSource during a unit test.
+    /// </summary>
+    internal class VerifyNoEventSourceErrorsLoggedTestAttribute : BeforeAfterTestAttribute
```

Comment:
Is this related to the PR?

Reply:
ConsoleExporter was failing silently. I was looking for a way to capture Errors logged to EventSource as that's our only indication that something went wrong. 
I think I can rewrite the test to catch the Exporter's exception now.

Longer term, I think there's value in having a mechanism to detect any other silent errors in the SDK.